src/datasets/iterable_dataset.py

```diff
@@ -3408,6 +3408,10 @@ def _concatenate_iterable_datasets(
     else:
         _check_column_names([col_name for dset in dsets for col_name in dset.features])
 
+    # Check format is consistent; if so, will set format for concatenated dataset
+    format_type_set = {dset._formatting.format_type for dset in dsets}
+    format_type = format_type_set.pop() if len(format_type_set) == 1 else None
```

Comment:
if the datasets have disparate formats, maybe you can add this logging info (same as for concatenating `Dataset` objects):
```python
logger.info("Some of the datasets have disparate format. Resetting the format of the concatenated dataset.")
```